test/Apache-NMS-AMQP-Interop-Test/NmsMessageProducerTest.cs

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Apache.NMS;
+using NUnit.Framework;
+
+namespace NMS.AMQP.Test
+{
+    [TestFixture]
+    public class NmsMessageProducerTest : AmqpTestSupport
+    {
+        [Test, Timeout(60_000)]
+        public void TestDeliveryDelay()
```

Comment:
This test cannot be accurately testing the feature, as the delivery support return capability must checked so that a client doesnt send it when a broker doesnt support it, 

like wise the delivery time on received message must = the senders time + delivery delay, thats persisteted in amqp message on send, and honoured by broker. 

lastly when a producer produces a message without delivery delay the time on received message must be the time the message is received NOT the time the calling client code asks for the value

Reply:
https://github.com/apache/qpid-jms/blob/198e4f2c7361ef4d62cc173d5f91ce24fea8e80b/qpid-jms-client/src/test/java/org/apache/qpid/jms/message/JmsMessagePropertyIntercepterTest.java#L1783

Reply:
https://github.com/apache/qpid-jms/blob/198e4f2c7361ef4d62cc173d5f91ce24fea8e80b/qpid-jms-client/src/test/java/org/apache/qpid/jms/integration/MessageIntegrationTest.java#L2271